### PR TITLE
Changes the type `mir::Mir` into `mir::Body`

### PR DESCRIPTION
**Pull request by TheSirC**
Fixes part 1 of #60229 (previously attempted in #60242).

I stumbled upon the issue and it seems that the previous attempt at solving it was not merged. This is a second try more up-to-date.

The commit should have changed comments as well.
At the time of writting, it passes the tidy and check tool.

### CODE REVIEW

File: src/librustc/mir/visit.rs [hunk 1]
```diff
@@ -71,7 +71,7 @@ macro_rules! make_mir_visitor {
             // Override these, and call `self.super_xxx` to revert back to the
             // default behavior.
 
-            fn visit_mir(&mut self, mir: & $($mutability)? Mir<'tcx>) {
+            fn visit_mir(&mut self, mir: & $($mutability)? Body<'tcx>) {
```
Comment (eddyb) [inline src/librustc/mir/visit.rs]: This and `super_mir` should be renamed to `visit_body`, and `super_body`, respectively.

File: src/test/run-pass/regions/regions-lub-ref-ref-rc.rs [hunk 2]
```diff
@@ -11,12 +11,12 @@
...
 
 #[derive(Clone)]
-enum CachedMir<'mir> {
+enum CachedBody<'mir> {
     Ref(&'mir String),
     Owned(Rc<String>),
 }
 
-impl<'mir> CachedMir<'mir> {
+impl<'mir> CachedBody<'mir> {
```
Comment (eddyb) [inline src/test/run-pass/regions/regions-lub-ref-ref-rc.rs]: No need to change this test - alternatively, `CachedMir` can be renamed to `Cached`, if you want.
(Since this is a reduction, where `Mir` was replaced with `String`)

**Review by eddyb — CHANGES_REQUESTED** [review]
r=me after a rebase and with nits fixed

File: src/test/run-pass/regions/regions-lub-ref-ref-rc.rs (hunk 2 above)
Comment (TheSirC) [reply src/test/run-pass/regions/regions-lub-ref-ref-rc.rs]: Well received. Renamed to `Cached` then !

File: src/librustc/mir/visit.rs (hunk 1 above)
Comment (TheSirC) [reply src/librustc/mir/visit.rs]: Rodger that. Done in the last force push.

**Comment by TheSirC** [comment]
@eddyb I rebased the branch with the modifications you asked to one commit.

**Review by eddyb — APPROVED** [review]

**Comment by eddyb** [comment]
@TheSirC You still need to rebase on latest master, before I can approve the PR.

**Comment by TheSirC** [comment]
Sorry for the back-and-forth, just rebased. I will try to be reactive in case another one is needed.

**Comment by TheSirC** [comment]
Obviously in the rebase I messed up an import; _normally fixed now_. Let's see what Travis has to say !

**Comment by TheSirC** [comment]
Should I rebase again ?

**Comment by TheSirC** [comment]
Rebased to fix conflicts. Tests were passing locally.

**Comment by TheSirC** [comment]
I have trouble piecing out if that error
   $ git checkout -qf  769df6dcad22f6c694324d115511eba1dd2f92c2
    fatal: reference is not a tree: 769df6dcad22f6c694324d115511eba1dd2f92c2
    The command "git checkout -qf 769df6dcad22f6c694324d115511eba1dd2f92c2" failed and exited with 128 during .

**Comment by TheSirC** [comment]
@eddyb or @Centril ? A retry ?

**Comment by eddyb** [comment]
@TheSirC Can you rebase?

**Comment by TheSirC** [comment]
Sure, I will push changes in the next 30 minutes.

**Comment by TheSirC** [comment]
I might have been a little optimistic about the deadline but now it's done! Sorry for the delay.

**Comment by eddyb** [comment]
No worries, you've waited enough for me too.

@bors r+ p=1 (mass rename that will easily bitrot)

**Comment by eddyb** [comment]
This is very weird (I'll start investigating myself, just to speed things up):
```
[00:35:05] thread 'rustc' panicked at 'attempt to subtract with overflow', src/libsyntax/source_map.rs:769:41
```

**Comment by eddyb** [comment]
I found a few more things to rename, btw (in comments/strings), if you want to do them:
(if not, they can be done in later PRs)
* these should say `mir::Body` (with backticks around it):
```
src/librustc/query/mod.rs
91:        /// Maps DefId's that have an associated Mir to the result

src/librustc/mir/visit.rs
264:                // than a for-loop, to avoid calling Mir::invalidate for

src/librustc_mir/transform/mod.rs
103:/// Where a specific Mir comes from.

src/librustc_mir/build/mod.rs
167:        // Convert the Mir to global types.
246:            // Convert the Mir to global types.

src/librustc/mir/interpret/mod.rs
50:    /// The index for promoted globals within their function's `Mir`.

src/librustc_mir/hair/cx/mod.rs
47:    /// See field with the same name on `Mir`.
```
* these should be "MIR" instead of "Mir".
```
src/librustc_mir/transform/qualify_consts.rs
1434:        tcx.sess.delay_span_bug(mir.span, "mir_const_qualif: Mir had errors");
1450:            tcx.sess.delay_span_bug(mir.span, "QualifyAndPromoteConstants: Mir had errors");
```

File: src/librustc_mir/transform/promote_consts.rs [hunk 3]
```diff
@@ -393,13 +393,13 @@ pub fn promote_candidates<'a, 'tcx>(mir: &mut Mir<'tcx>,
         }
 
 
-        // Declare return place local so that `Mir::new` doesn't complain.
+        // Declare return place local so that `Body::new` doesn't complain.
```
Comment (eddyb) [inline src/librustc_mir/transform/promote_consts.rs]: This should say `mir::Body::new`.

File: src/librustc_mir/borrow_check/location.rs [hunk 4]
```diff
@@ -1,4 +1,4 @@
-use rustc::mir::{BasicBlock, Location, Mir};
+use rustc::mir::{BasicBlock, Location, Body};
```
Comment (eddyb) [inline src/librustc_mir/borrow_check/location.rs]: I don't remember if @oli-obk wanted to leave this for another PR (or maybe I never asked for this before), but IMO `Body` shouldn't be referred to as anything other than `mir::Body` outside of the module it's defined in.

File: src/librustc/ty/steal.rs [hunk 5]
```diff
@@ -9,7 +9,7 @@ use rustc_data_structures::sync::{RwLock, ReadGuard, MappedReadGuard};
 /// queries are these pure functions that produce an immutable value
 /// (since if you did the query twice, you could observe the
 /// mutations). So instead we have the query produce a `&'tcx
-/// Steal<Mir<'tcx>>` (to be very specific). Now we can read from this
+/// Steal<Body<'tcx>>` (to be very specific). Now we can read from this
```
Comment (eddyb) [inline src/librustc/ty/steal.rs]: This should say `mir::Body` instead of `Body`.

**Comment by eddyb** [comment]
Oh wow, the failure has nothing to do with this PR, it just accidentally got triggered:
https://github.com/rust-lang/rust/blob/721268583759224d0f6476e0b8b196cc8afbdea0/src/libsyntax/source_map.rs#L725-L731

That code is missing a check that `local_begin` and `local_end` *actually are in the same `SourceFile`*, like this (from a different function):
https://github.com/rust-lang/rust/blob/721268583759224d0f6476e0b8b196cc8afbdea0/src/libsyntax/source_map.rs#L514-L517

Macros are involved, somehow causing a span to end up with its two endpoints in different files. That might otherwise result in ICEs, if not for them being *at the exact same* byte offset (within their respective files) which the algorithm doesn't account for.
cc @davidtwco @petrochenkov

**Comment by eddyb** [comment]
@TheSirC I've pushed two commits - if you need to rebase, please pull first (and don't squash the `syntax: ...` commit away, since it's a separate bugfix).

@bors r+
